### PR TITLE
create tests for all api Coworking.

### PR DESCRIPTION
**Pull request by Lex-23**
- create TestCases for Api Coworkings (Views for work with this model)
- update this Api, validators and permissions after this

### CODE REVIEW

File: coworkings_be/coworking/tests/test_api.py [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+        )
+        self.url = reverse("coworking-list")
+        self.maxDiff = None
+
+    def test_get_list_coworking(self):
+        response = self.client.get(self.url)
+        data = self.coworking_list
+        self.assertEqual(200, response.status_code)
+        self.assertEqual(len(data), response.json()["count"])
+        for n in range(len(data)):
```
Comment (andreyPromaster) [inline coworkings_be/coworking/tests/test_api.py:30]: You can use zip function:
`for expected_date, actual_data in zip(data, response.json()["results"])`

File: coworkings_be/users/tests/factory.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+class UserFactory(factory.django.DjangoModelFactory):
+    class Meta:
+        model = User
+
+    email = factory.Faker("email")
+    role = UserRoles.OWNER
+
+
+class TestUsersMixin:
```
Comment (andreyPromaster) [inline coworkings_be/users/tests/factory.py:16]: In general, it  does not need to create different 3 users for test, when user is needed

File: coworkings_be/coworking/tests/test_api.py [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+        user_owner = self.test_users["owner"]
+
+        data = {"title": "New coworking", "city": "Minsk"}
+        token = CustomTokenObtainPairSerializer.get_token(user_owner)
+        self.client.force_authenticate(user=user_owner, token=token)
+        response = self.client.post(self.url, data=data)
+        new_coworking = Coworking.objects.get(id=response.json()["id"])
+
+        self.assertEqual(201, response.status_code)
+        self.assertEqual(data["title"], response.json()["title"])
```
Comment (andreyPromaster) [inline coworkings_be/coworking/tests/test_api.py:80]: By the way, you can store the expected json in a file and compare the actual response with the real response.

File: coworkings_be/users/tests/factory.py (hunk 2 above)
Comment (Lex-23) [reply coworkings_be/users/tests/factory.py:16]: But I need this three roles in more then one test cases. I catch your point. Maybe we need something like factory for each role

File: coworkings_be/coworking/tests/test_api.py (hunk 3 above)
Comment (Lex-23) [reply coworkings_be/coworking/tests/test_api.py:80]: but expected json will be different for different test cases. Too much code in my opinion

File: coworkings_be/users/tests/factory.py (hunk 2 above)
Comment (Lex-23) [reply coworkings_be/users/tests/factory.py:16]: I think, I need this mixin only for cases, when requirement all three roles

**Review by andreyPromaster — APPROVED** [review]

**Pull request merged**